### PR TITLE
first variant

### PR DESCRIPTION
**Pull request by voloshynamaryna11**

### CODE REVIEW

**Review by KostiantynVorobiov — APPROVED** [review]

File: src/main/java/internet/market/dao/impl/OrderDaoImpl.java [hunk 1]
```diff
@@ -36,6 +36,11 @@ public Order createOrder(Order order) {
         return Storage.orders;
     }
 
+    @Override
+    public Order update(Order item) {
+        return item;
```
Comment (Sofasmile) [inline src/main/java/internet/market/dao/impl/OrderDaoImpl.java:45]: let's implement this method

File: src/main/java/internet/market/dao/impl/ShoppingCartDaoImpl.java [hunk 2]
```diff
@@ -15,15 +16,36 @@ public ShoppingCart create(ShoppingCart shoppingCart) {
...
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        return Storage.shoppingCarts;
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart item) {
+        return item;
```
Comment (Sofasmile) [inline src/main/java/internet/market/dao/impl/ShoppingCartDaoImpl.java:37]: the same

File: src/main/java/internet/market/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -22,14 +22,19 @@ public Order completeOrder(ShoppingCart shoppingCart) {
...
     }
 
     @Override
     public List<Order> getUserOrders(Long userId) {
         return orderDao.getUserOrders(userId);
     }
 
+    @Override
+    public Order create(Order item) {
+        return item;
```
Comment (Sofasmile) [inline src/main/java/internet/market/service/impl/OrderServiceImpl.java]: the same

File: src/main/java/internet/market/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -40,6 +45,11 @@ public Order get(Long id) {
         return orderDao.getAll();
     }
 
+    @Override
+    public Order update(Order item) {
+        return item;
```
Comment (Sofasmile) [inline src/main/java/internet/market/service/impl/OrderServiceImpl.java]: )

File: src/main/java/internet/market/dao/ShoppingCartDao.java [hunk 5]
```diff
@@ -3,10 +3,8 @@
...
 import java.util.Optional;
 
-public interface ShoppingCartDao {
-    ShoppingCart create(ShoppingCart shoppingCart);
-
+public interface ShoppingCartDao extends GenericDao<ShoppingCart, Long> {
     Optional<ShoppingCart> getByUserId(Long userId);
 
-    boolean delete(ShoppingCart shoppingCart);
+    boolean deleteByObject(ShoppingCart shoppingCart);
```
Comment (Sofasmile) [inline src/main/java/internet/market/dao/ShoppingCartDao.java]:
```suggestion
    boolean delete(ShoppingCart shoppingCart);
```

File: src/main/java/internet/market/dao/impl/OrderDaoImpl.java (hunk 1 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/dao/impl/OrderDaoImpl.java:45]: done

File: src/main/java/internet/market/dao/impl/ShoppingCartDaoImpl.java (hunk 2 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/dao/impl/ShoppingCartDaoImpl.java:37]: done

File: src/main/java/internet/market/service/impl/OrderServiceImpl.java (hunk 4 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/service/impl/OrderServiceImpl.java]: done

File: src/main/java/internet/market/service/impl/OrderServiceImpl.java (hunk 3 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/service/impl/OrderServiceImpl.java]: done

File: src/main/java/internet/market/dao/ShoppingCartDao.java (hunk 5 above)
Comment (voloshynamaryna11) [reply src/main/java/internet/market/dao/ShoppingCartDao.java]: done

File: src/main/java/internet/market/dao/impl/ShoppingCartDaoImpl.java [hunk 6]
```diff
@@ -38,15 +42,15 @@ public boolean delete(Long id) {
         return Storage.shoppingCarts.removeIf(shoppingCart -> shoppingCart.getId().equals(id));
     }
 
+    public boolean delete(ShoppingCart shoppingCart) {
+        return Storage.shoppingCarts
+                .removeIf(shoppingCartUnit -> shoppingCartUnit.equals(shoppingCart));
```
Comment (Sofasmile) [inline src/main/java/internet/market/dao/impl/ShoppingCartDaoImpl.java]:
```suggestion
        return Storage.shoppingCarts.remove(shoppingCart);
```

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**